### ISSUE TITLE
Inheritance 3

### ISSUE THREAD

**Issue by Udayv56** [Udayv56/CPP#25]
The `protected` access specifier is a visibility mode in object-oriented programming that allows the members of a class to be accessible within the class itself and its derived classes. The protected members are not accessible outside the class or in unrelated classes.

**CODE-**

#include<iostream>
using namespace std;

class Base{
    
    protected: // it is a kind of private variable which can be inherite
        int a;
    private:
        int b;
    
    
};
/* Inheritance:
                 public          private        protected (derived)
1.Private       Not inherited   Not inherited   Not inherited
2.protected     Protected       Private         Protected           
3.public        public          Private         Protected

*/
class Derived : protected Base{
    
};

int main(){
    Base b;
    Derived d;
    // cout<<b.a; not allowed
    // cout<<d.a; not allowed a is still protected
    
    return 0;
}

Now let's analyze the provided code:

1. The code begins with the inclusion of the necessary header file, `<iostream>`, for input/output operations.

2. The `Base` class is defined, which has two data members: `a` (protected) and `b` (private).

3. Inside the `Base` class, `a` is declared as protected. This means that it is accessible within the class and any derived classes.

4. `b` is declared as private, which means it is only accessible within the `Base` class itself.

5. The `Derived` class is defined, which is derived from the `Base` class using the protected access specifier. This means that the protected members of the `Base` class will become protected members of the `Derived` class.

6. In the `main()` function, an object `b` of the `Base` class and an object `d` of the `Derived` class are created.

7. Since `a` is declared as protected in the `Base` class, it is not directly accessible outside the class or in unrelated classes. Therefore, attempting to print `b.a` or `d.a` will result in a compilation error.

8. The program ends with a `return 0` statement.

In summary, the code demonstrates the usage of the `protected` access specifier. It shows that protected members are accessible within the class itself and its derived classes, but not accessible outside the class or in unrelated classes.
